File: src/main/java/com/kylexu/leetcode/matrix/Solution2500.scala
package com.kylexu.leetcode.matrix

object Solution2500 {

  def deleteGreatestValue(grid: Array[Array[Int]]): Int = {
    // todo
    val rs = grid.map(_.sortWith(_ > _))
      .sortWith {
        (a, b) =>
          a.zip(b).collectFirst {
            case (x, y) if (x != y) => x > y
          }.getOrElse(false)
      }

    var sum = 0
    var i = 0

    for (elem <- grid.indices) {
      sum += grid(elem)(i)
      i += 1
    }

    sum
  }

  // grid = [[1,2,4],[3,3,1]]
  def main(args: Array[String]): Unit = {
    val grid = Array(
      Array(1, 2, 4),
      Array(3, 3, 1)
    )
    val rs = Solution2500.deleteGreatestValue(grid)
    println(rs)
  }
}

File: src/main/java/com/kylexu/leetcode/matrix/Solution2500.java
package com.kylexu.leetcode.matrix;

import java.util.Arrays;

public class Solution2500 {
    public static void main(String[] args) {
        int[][] grid = {{1, 2, 4}, {3, 3, 1}};
        int i = new Solution2500().deleteGreatestValue(grid);
        System.out.println("i = " + i);

    }

    public int deleteGreatestValue(int[][] grid) {
        for (int[] ints : grid) {
            Arrays.sort(ints);
        }

        int rs = 0;
        int m = grid.length;
        int n = grid[0].length;

        for (int i = 0; i < n; i++) {
            int max = 0;
            for (int j = 0; j < m; j++) {
                max = Math.max(max, grid[j][i]);
            }
            rs += max;
        }

        return rs;
    }
}
